torch/fx/graph.py

```diff
@@ -130,6 +167,7 @@ def node_copy(self, node: Node, arg_transform: Callable[[Node], Argument] = lamb
     def output(self, result: Argument):
         self.result = result
         self._mark_uses(result)
+        self.nodes.finalize()
```

Comment:
btw, have you considered just doing `self.nodes = tuple(self.nodes)`? :) What is the benefit of FinalizableList over Tuple?

Reply:
Every time you append to the tuple you'd have to create a new data structure? Also that doesn't match with the specification above, of immutable after an output has been specified

Reply:
I mean to change type from `list` to `tuple` only when you want to finalize. Afaiu, list and tuple are mostly indistinguishable from the consumption standpoint.